cpp/include/cugraph/algorithms.hpp

```diff
@@ -1720,50 +1720,63 @@ k_core(raft::handle_t const& handle,
  * @tparam edge_t Type of edge identifiers. Needs to be an integral type.
  * @tparam weight_t Type of edge weights. Needs to be a floating point type.
  * @tparam edge_type_t Type of edge type. Needs to be an integral type.
+ * @tparam label_t Type of label. Needs to be an integral type.
  * @tparam store_transposed Flag indicating whether sources (if false) or destinations (if
  * true) are major indices
  * @tparam multi_gpu Flag indicating whether template instantiation should target single-GPU (false)
  * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
  * handles to various CUDA libraries) to run graph algorithms.
  * @param graph_view Graph View object to generate NBR Sampling on.
  * @param edge_weight_view Optional view object holding edge weights for @p graph_view.
- * @param edge_id_type_view Optional view object holding edge ids and types for @p graph_view.
+ * @param edge_id_view Optional view object holding edge ids for @p graph_view.
+ * @param edge_type_view Optional view object holding edge types for @p graph_view.
  * @param starting_vertices Device vector of starting vertex IDs for the sampling.
  * @param starting_labels Optional device vector of starting vertex labels for the sampling.
+ * @param starting_vertex_offsets Optional device vector organizing @p starting_vertices into
+ * batches of vertices, used in place of @p starting_labels.  If @p starting_vertex_offsets is
+ * specified then output will include offsets element (defining the offsets that correspond to a
+ * particular input batch)
```

Comment:
See the above discussion.  I did not intend this parameter to be what you perhaps are imagining.  I think we should put together some example input/output samples of what we think we want in different scenarios and review these (including @alexbarghi-nv and @VibhuJawa in that review) so that we can all agree on what is desired here.  I'll try to put together some examples for review and schedule something for next week.

Reply:
I assume this discussion is no longer relevant if we use `starting_vertex_labels` (size == `starting_vertices.size()`).